core/src/Kokkos_Parallel_Reduce.hpp

```diff
@@ -71,11 +71,7 @@ struct ParallelReduceReturnValue<
   using return_type  = ReturnType;
   using reducer_type = InvalidType;
 
-  using value_type_scalar = typename return_type::value_type;
-  using value_type_array  = typename return_type::value_type* const;
-
-  using value_type = std::conditional_t<return_type::rank == 0,
-                                        value_type_scalar, value_type_array>;
```

Comment:
Array reductions require specifying `value_type` anyway which overrides the value type declared here in `FunctorAnalysis` so we might as well just always use `return_type::value_type`.

Reply:
What I mean is that the type is unused when the functor defines `value_type` and that is requires for array reductions. In other words, the rank of the view doesn't decide if we do a array reduction or not but the functor's `value_type` is. When looking at the issue, I was confused by this type in particular and thought removong it makes it clearer.

Reply:
If others think it's useful to keep, I'm happy to put it back.

Reply:
thanks for the explanation. AFAIKT we should be good